### PR TITLE
Adding @Expose attribute to requests, responses and error responses in support of logging

### PR DESCRIPTION
**Pull request by shoatman**
Added additional non-PII logging of requests and response objects.  Controlling access to PII data via the Expose annotation.  Data that is considered PII is not marked with the attribute.  Then Gson is configured to only serialize fields that are marked with the annotation.  Regular Gson instances are unaffected by this attribute.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java [hunk 1]
```diff
@@ -58,12 +59,14 @@
     /**
      * Can be used to pre-fill the username/email address field of the sign-in page for the user, if you know their username ahead of time.
      */
+    @Expose()
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: PII?

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/ISuccessResponse.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.providers.oauth2;
+
+public interface ISuccessResponse {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/ISuccessResponse.java]: This looks like a marker interface but it's not immediately clear to me how it's being used - is this being consumed somewhere?

**Comment by iambmelt** [comment]
As a minor syntactic note, we should be able to omit the empty parens and tag fields as `@Expose` (instead of `@Expose()`)

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/ISuccessResponse.java (hunk 2 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/ISuccessResponse.java]: Yes.  IResult was updated.

File: common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java (hunk 1 above)
Comment (shoatman) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: Wasn't sure about this myself.  Am expecting a document that may shed some additional light.

Comment (heidijinxujia) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAuthorizationRequest.java]: upn and login hint are pii.

File: common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenOperationParameters.java [hunk 3]
```diff
@@ -33,9 +34,13 @@
 public class AcquireTokenOperationParameters extends OperationParameters {
 
     private transient Activity mActivity;
+    @Expose()
```
Comment (heidijinxujia) [inline common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenOperationParameters.java]: pii

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationRequest.java [hunk 4]
```diff
@@ -66,6 +69,7 @@
      * application, the authorization server will redirect the user back to the application with
      * either an authorization code or access token in the URL.
      */
+    @Expose()
```
Comment (heidijinxujia) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationRequest.java]: @shoatman redirect_uri is OII. Not sure if we should include it into logging.

**Pull request merged**